ESP8266/ESP8266.cpp

```diff
@@ -181,7 +182,13 @@ int8_t ESP8266::getRSSI()
 
 bool ESP8266::isConnected(void)
 {
-    return getIPAddress() != 0;
+    const char *ip_buff = getIPAddress();
+
+    if(!ip_buff || std::strcmp(ip_buff, "0.0.0.0") == 0) {
+        return 0;
```

Comment:
Prefer false since this is returning bool.

Reply:
Good point